### PR TITLE
[ESIMD] Add a test specialization constant that is set but not used

### PR DESCRIPTION
**Pull request by v-klochkov**
Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

**Comment by v-klochkov** [comment]
This test verifies the fix merged here: https://github.com/intel/llvm/pull/6464

**Review by fineg74 — DISMISSED** [review]

File: SYCL/ESIMD/unused_spec_const.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     CGH.set_specialization_constant<Spec>(42);
+     CGH.single_task([=](sycl::kernel_handler KH) SYCL_ESIMD_KERNEL {
+       // This use of get_specializaiton_constant is commented out.
+       // Thus the device code doen't have info about it and cannot allocate
+       // memory/buffer for it on device side.
+       // The test checks that the unused kernel parameter is either
+       // eliminated by DAE optimization OR nullptr is passed for that argument.
+
+       // KH.get_specialization_constant<Spec>();
+       *A += 2;
```
Comment (kbobrovs) [inline SYCL/ESIMD/unused_spec_const.cpp]: formally speaking, we don't support C++ style memory operations in ESIMD (even though they sometimes work)

Comment (v-klochkov) [reply SYCL/ESIMD/unused_spec_const.cpp:36]: wait, it is not memory operation. It is "*A = *A + 2;"

Comment (v-klochkov) [reply SYCL/ESIMD/unused_spec_const.cpp]: Thank you. I replaced it with ESIMD load/store: 30ab516311043d6a989f3a03200d7048b00d5af0

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**